osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -116,6 +120,12 @@ protected override bool OnStart()
                 Mods.Value = mods;
             }
 
+            // Default to local leaderboard if the currently selected leaderboard doesn't have scores or is unavailable
+            // perhaps because it requires sign in, requires the beatmap to be ranked, etc.
+            bool isLeaderboardSelected = config.Get<PlayBeatmapDetailArea.TabType>(OsuSetting.BeatmapDetailTab) != PlayBeatmapDetailArea.TabType.Details;
+            if (isLeaderboardSelected && !playBeatmapDetailArea.Leaderboard.HasScores())
+                config.SetValue(OsuSetting.BeatmapDetailTab, PlayBeatmapDetailArea.TabType.Local);
```

Comment:
Rather than `.Get<>()` and then `.SetValue()`, I'd just get `config.GetBindable<>()`, store that to a field, and then read from/write to the setting via the bindable.